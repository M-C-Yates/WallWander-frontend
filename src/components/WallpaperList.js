import React, { useEffect, useState } from 'react';
import axios from 'axios';
import ReactPaginate from 'react-paginate';
import Image from 'react-lazy-image';
import Masonry from 'react-masonry-component';
import { useDispatch } from 'react-redux';
import { Link } from '@reach/router';

import { getImages } from '../actions/image';
import { baseUrl } from '../App';

export const WallpaperList = (props) => {
	const [images, setImages] = useState([]);
	const [pageCount, setPageCount] = useState(0);

	const dispatch = useDispatch();

	useEffect(() => {
		const fetchData = async () => {
			const { data } = await axios.get(`${baseUrl}/images?limit=5&page=0`);
			const images = data.data;
			dispatch(getImages(images));
			setImages(images);
			setPageCount(data.last_page);
			return;
		};
		fetchData();
		return;
	}, [dispatch]);

	const pageChange = async ({ selected }) => {
		const { data } = await axios.get(
			`${baseUrl}/images?limit=5&page=${selected}`
		);
		const images = data.data;

		dispatch(getImages(images));
		setImages(images);
		setPageCount(data.last_page);
		return;
	};

	return (
		<section>
			<p>wallpapers!</p>
			<Masonry>
				{images.map((image) => {
					const urlArr = image.secureUrl.split('/');
					urlArr[6] = 'f_auto,c_fit,h_400,w_500';
					image.secureUrl = urlArr.join('/');
					return (
						<Link to={`/wallpapers/${image.id}`} key={image.title}>
							<Image source={image.secureUrl} alt={image.title} />
						</Link>
					);
				})}
			</Masonry>

			{pageCount > 1 ? (
				<ReactPaginate
					pageCount={pageCount}
					pageRangeDisplayed={5}
					marginPagesDisplayed={3}
					onPageChange={pageChange}
				/>
			) : (
				<React.Fragment />
			)}
		</section>
	);
};
