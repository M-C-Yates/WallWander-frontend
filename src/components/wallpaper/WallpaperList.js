import React, { useEffect, useState } from "react";
import axios from "axios";
import ReactPaginate from "react-paginate";
import Thumb from "./Thumb";

import { useDispatch } from "react-redux";
import { getImages } from "../../actions/image";

export const WallpaperList = (props) => {
  const { images, setImages, pageChangeUrl, paginationData } = props;
  const [pageCount, setPageCount] = useState(0);

  const dispatch = useDispatch();

  useEffect(() => {
    setPageCount(paginationData.lastPage);
    return;
  }, [paginationData.lastPage]);

  const pageChange = async ({ selected }) => {
    const { data } = await axios.get(
      `${pageChangeUrl}?limit=15&page=${selected}`
    );
    const images = data.data;

    dispatch(getImages(images));
    setImages(images);
    setPageCount(data.last_page);
    return;
  };

  const columnWidth = window.innerWidth - 27 <= 700 ? 250 : 500;

  return (
    <React.Fragment>
      <div className="wallpaperslist">
        {images.map((image) => {
          const urlArr = image.secureUrl.split("/");
          urlArr[6] = "f_auto,h_450,w_500,c_limit";
          image.secureUrl = urlArr.join("/");
          return <Thumb image={image} key={image.id} />;
        })}
      </div>

      {pageCount > 1 ? (
        <ReactPaginate
          pageCount={pageCount}
          pageRangeDisplayed={columnWidth <= 700 ? 2 : 3}
          marginPagesDisplayed={1}
          onPageChange={pageChange}
          activeClassName="pagination--active"
          activeLinkClassName="pagination--active-link text-purple-600 border-b border-purple-600"
          breakClassName="pagination--break"
          breakLinkClassName="pagination--break-link"
          containerClassName="pagination text-4xl"
          disabledClassName="pagination--disabled invisible"
          nextClassName="pagination--next"
          nextLinkClassName="pagination--next-link"
          pageClassName="pagination--page mx-4 "
          pageLinkClassName="pagination--page-link hover:text-purple-600"
          previousClassName="pagination--previous"
          previousLinkClassName="pagination--previous-link"
        />
      ) : (
        <React.Fragment />
      )}
    </React.Fragment>
  );
};

export default WallpaperList;
