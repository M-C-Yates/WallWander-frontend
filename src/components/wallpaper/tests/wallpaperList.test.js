import React from 'react';
import { render, cleanup } from '@testing-library/react';

import WallpaperList from '../WallpaperList';

afterEach(cleanup);

jest.mock('react-redux', () => {
	return {
		useDispatch() {
			return { dispatch: jest.fn() };
		},
		useSelector() {
			return true;
		}
	};
});

describe('WallpaperList', () => {
	const mock = jest.fn();
	const props = {
		images: [],
		setImages: mock,
		pageChangeUrl: '/123/',
		paginationData: {
			lastPage: 12
		}
	};
	it('should match snapshot', () => {
		const { asFragment } = render(
			<WallpaperList
				images={props.images}
				setImages={props.setImages}
				pageChangeUrl={props.pageChangeUrl}
				paginationData={props.paginationData}
			/>
		);
		expect(asFragment).toMatchSnapshot();
	});
});
