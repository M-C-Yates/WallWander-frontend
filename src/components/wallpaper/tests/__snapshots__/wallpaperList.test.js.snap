// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WallpaperList should match snapshot 1`] = `[Function]`;
